crates/provider-wit-bindgen/src/lib.rs

```diff
@@ -504,29 +1317,42 @@ impl WitBindgenOutputVisitor {
 
 impl VisitMut for WitBindgenOutputVisitor {
     fn visit_item_mod_mut(&mut self, node: &mut ItemMod) {
-        // Save the WIT namespace that we've recognized
+        #[cfg(feature = "debug")]
+        eprintln!(
+            "[info] (bindgen module hierarchy): {}> {}",
+            "=".repeat(self.current_module_level()),
+            node.ident
+        );
+
+        // Detect the WIT namespace while traversing the bindgen output
+        //
+        // We expect the top level rust module (i.e. level zero of the module hierarchy)
+        // in a package with imports to be the wit namespace.
         //
-        // The top level WIT namespace is always a module at @ level zero
-        // of the item hierarchy in packages with imports, but is @ level 1 in packages
-        // with only exports
-        if (self.current_module_level() == 0 && node.ident != EXPORTS_MODULE_NAME)
-            || (self.current_module_level() == 1 && self.at_exported_module())
+        // Packages with exports (which may *only* have exports) have 'exports' at level zero
+        // then follow a similar pattern
+        if self.wit_ns.is_none()
+            && ((self.current_module_level() == 0 && node.ident != EXPORTS_MODULE_NAME)
+                || (self.current_module_level() == 1 && self.at_exported_module()))
         {
             self.wit_ns = Some(node.ident.to_string());
         }
 
-        // Save the WIT package name
+        // Detect the WIT package while traversing the bindgen output
         //
-        // In packages with imports, the package name will be at level 1 (under the ns)
-        // but in packages with only exports, it will be at level 2 under 'exports' and the ns)
+        // We expect the second level rust module (i.e. level 1 in a zero-indexed module hierarchy)
+        // in a package with imports to be the wit package name.
+        //
+        // Packages with exports (which may *only* have exports) have 'exports' at level zero
```

Comment:
This comment says level zero, but the code below uses level 2. Am I reading this right?

Reply:
yes -- `exports` is at level zero -- but the actual pakage is at level 2 (i.e. `exports -> <namespace> -> <package>`)

maybe this is a bit confusing, I'll clarify

- [ ] clarify comment about exports